_posts/2023-01-02-automating-data-workflow-stepfunctions.md

```diff
@@ -0,0 +1,58 @@
+---
+layout: post
+font: serif
+title: "Automating Data Workflows with AWS Step Functions: An Example Using Fivetran and dbt"
+date: 2023-01-03
+highlight: monokai
+author: Joseph Miocevich
+image: 
+tags: stepfunction aws fivetran lambda dbt
+---
+
+
+Automating and orchestrating data workflows can be a daunting task, but with the power of AWS Step Functions, it's easier than you might think! In this post, we'll show you how to use Step Functions to coordinate the execution of multiple AWS services and streamline your workflows with Fivetran connectors and dbt. 
+
+# Setting the Stage
+
+Before we dive in, make sure you have the following:
+
+- An AWS account
+- Familiarity with Fivetran and dbt
+
+Solution Overview
+
+To give you an idea of how everything fits together, here's a diagram of the solution architecture:
+
+![architecture diagram](/img/step_functions_fivetran.png)
+
+As you can see, the solution consists of:
+
+- An Amazon Elastic Container Registry (ECR) repository for storing the Docker image for the ECS task
+Three AWS Lambda functions:
+- `getConnectorList`, which retrieves a list of connectors from Fivetran
+- `syncFivetranConnectors`, which synchronizes a single Fivetran connector
+- `fivetranWebook`, which is triggered by a webhook from Fivetran and starts a new execution of the state machine
+- An AWS HttpApi endpoint for exposing the fivetranWebook function
+- A DynamoDB table for storing [wait task token](https://docs.aws.amazon.com/step-functions/latest/dg/callback-task-sample-sqs.html) for the state machine
+- An Amazon Elastic Container Service (ECS) cluster for running the ECS task
+- An AWS Step Functions state machine for orchestrating the connectors and running the ECS task
+
+# AWS Step Functions in Action
+
+AWS Step Functions can be used to build a workflow that automates tasks in your system. In this particular solution, a state machine is defined in the statemachine.yml file and outlines the steps in the workflow. The state machine starts by calling a Lambda function called getConnectorList, which retrieves a list of connectors from  Fivetran and stores the result in the state machine's output.
```

Comment:
is there an example of the getConnectorList output? Just feels a bit light..

Reply:
sure